homeassistant/components/light/mqtt.py

```diff
@@ -121,10 +122,17 @@
     """Set up MQTT light dynamically through MQTT discovery."""
     async def async_discover(discovery_payload):
         """Discover and add a MQTT light."""
-        config = PLATFORM_SCHEMA(discovery_payload)
-        await _async_setup_entity(hass, config, async_add_entities,
-                                  discovery_payload[ATTR_DISCOVERY_HASH])
-
+        if discovery_payload['platform'] == 'mqtt_json':
+            mqtt_json = get_platform(hass, 'light', 'mqtt_json')
```

Comment:
this is wrong. Platforms should not know about other platforms.

We have two choices:
 - don't support `mqtt_json` with config entries
 - move MQTT JSON into the MQTT platform and add a config to switch to "mode". To keep code manageable we could move the files so we have `light/mqtt/__init__.py` and `light/mqtt/json.py`

Reply:
Hmm, too bad.
There's the third light variant: `mqtt_template`, which has the same issue then.

When you say, "move MQTT JSON into the MQTT platform", does that affect the `configuration.yaml` entry?

``` yaml
light:
  - platform: mqtt_json
    command_topic: "home/rgb1/set"
```

``` yaml
light:
  - platform: mqtt_template
    command_topic: "home/rgb1/set"
    command_on_template: "on"
    command_off_template: "off"
```